docs/docs/reference/metaprogramming/macros-more-examples.md

```diff
@@ -0,0 +1,79 @@
+---
+title: "Macros: More Examples"
```

Comment:
All these examples should probably go in https://docs.scala-lang.org/scala3/guides/macros/faq.html and we should add some references to the tutorial and FAQ in `macros.md` and `reflection.md`.